vowpalwabbit/core/src/reductions/stagewise_poly.cc

```diff
@@ -686,14 +678,49 @@ base_learner* VW::reductions::stagewise_poly_setup(VW::setup_base_i& stack_build
   poly->original_ec = nullptr;
   poly->next_batch_sz = poly->batch_sz;
 
-  auto* l = VW::LEARNER::make_reduction_learner(std::move(poly), as_singleline(stack_builder.setup_base_learner()),
-      learn, predict, stack_builder.get_setupfn_name(stagewise_poly_setup))
-                .set_input_label_type(VW::label_type_t::SIMPLE)
-                .set_output_prediction_type(VW::prediction_type_t::SCALAR)
-                .set_save_load(save_load)
-                .set_finish_example(::finish_example)
-                .set_end_pass(end_pass)
-                .build();
+  auto* l =
+      VW::LEARNER::make_reduction_learner(std::move(poly), as_singleline(stack_builder.setup_base_learner()), learn,
+          predict, stack_builder.get_setupfn_name(stagewise_poly_setup))
+          .set_input_label_type(VW::label_type_t::SIMPLE)
+          .set_output_prediction_type(VW::prediction_type_t::SCALAR)
+          .set_save_load(save_load)
+          .set_output_example_prediction(VW::details::output_example_prediction_simple_label<stagewise_poly>)
+          .set_update_stats(
```

Comment:
Is there a reason to make these lambdas? Seems less compact than the usual design